gst/nnstreamer/tensor_common.c

```diff
@@ -1538,24 +1575,43 @@ gst_tensor_meta_info_validate (GstTensorMetaInfo * meta)
   g_return_val_if_fail (meta != NULL, FALSE);
   g_return_val_if_fail (GST_TENSOR_META_VERSION_VALID (meta->version), FALSE);
 
-  if (meta->type >= _NNS_END)
-    return FALSE;
+  if (meta->type >= _NNS_END) {
+    nns_logw ("Failed to validate tensor meta info. type: %s. ",
+        GST_STR_NULL (gst_tensor_get_type_string (meta->type)));
+    goto error;
+  }
 
   for (i = 0; i < NNS_TENSOR_META_RANK_LIMIT; i++) {
     if (meta->dimension[i] == 0) {
-      if (i == 0)
-        return FALSE;
+      if (i == 0) {
+        gchar *dim_str = gst_tensor_get_dimension_string (meta->dimension);
+        nns_logw ("Failed to validate tensor meta info. Given dimension: %s",
+            dim_str);
+        g_free (dim_str);
+        goto error;
+      }
       break;
     }
   }
 
-  if (meta->format >= _NNS_TENSOR_FORMAT_END)
-    return FALSE;
+  if (meta->format >= _NNS_TENSOR_FORMAT_END) {
+    nns_logw ("Failed to validate tensors meta info. format: %s. ",
+        GST_STR_NULL (gst_tensor_get_format_string (meta->format)));
+    goto error;
+  }
 
-  if (meta->media_type > _NNS_TENSOR)
-    return FALSE;
+  if (meta->media_type > _NNS_TENSOR) {
+    nns_logw ("Failed to validate tensor meta info. invalid media type: %d.",
+        meta->media_type);
+    goto error;
+  }
 
   return TRUE;
+
+error:
+  nns_logw
+      ("NNStreamer internal error. Please post issue on https://github.com/nnstreamer/nnstreamer/issues.");
+  return FALSE;
```

Comment:
Be careful. You don't know if returning FALSE is an issue or not. The caller knows if FALSE is an issue or not.